handshake.go

```diff
@@ -583,7 +583,7 @@ func (t *handshakeTransport) enterKeyExchange(otherInitPacket []byte, serverInfo
 	}
 	// get server information needed for SSH scanner
 	// need to asign before return of error because keyCallBack always returns an error
-	if serverInfo != nil {
+	if serverInfo != nil && hostKey != nil {
```

Comment:
Are there circumstances where we get a `serverInit` but no `hostKey`?

Reply:
Yes, at least the scanner through a panic for this exact reason on some IP addresses

Reply:
So we might want to store the serverInfo without the hostKey?

Can you provide this IP address for testing purposes?

Reply:
If the serverInfo struct is filled, I think it would make sense to store that information despite not having the key.
I will have to scan the 10.000 IPs with one scanner instance to get the Address. So yes I can provide it, but I need to search for it first.

Reply:
So splitting the combined `if` into two separate `if`s?